Source/ImPlotWidgets/Private/LineFilledGraphWidget.cpp

```diff
@@ -17,24 +19,40 @@ void SFilledGraphWidget::Construct(const FArguments& InArgs)
 			.OnImGuiDrawDelegate(this, &SFilledGraphWidget::OnImGuiDraw)
 		];
 
-	HandleDataArray();
+	HandleDataArrays();
 }
 
-void SFilledGraphWidget::HandleDataArray()
+void SFilledGraphWidget::HandleDataArrays()
 {
-	Algo::Sort(InData2D, [](const FDataPoint& LeftDataPoint, const FDataPoint& RightDataPoint)
-		{
-			return LeftDataPoint.X < RightDataPoint.X;
-		});
+	XMaxValue = DataArrays[0][0].X;
+	YMaxValue = DataArrays[0][0].Y;
+	YMinValue = 0.0f;
 
-	for (FDataPoint Point : InData2D)
+	for (TArray<FDataPoint> CurrentDataArray : DataArrays)
 	{
-		XValues.Add(Point.X);
-		YValues.Add(Point.Y);
-	}
+		Algo::Sort(CurrentDataArray, [](const FDataPoint& LeftDataPoint, const FDataPoint& RightDataPoint)
+			{
+				return LeftDataPoint.X < RightDataPoint.X;
+			});
+
+		TArray<float> XValues;
+		TArray<float> YValues;
+
+		for (FDataPoint Point : CurrentDataArray)
+		{
+			XValues.Add(Point.X);
+			YValues.Add(Point.Y);
+		}
+
+		XValueArrays.Add(XValues);
+		YValueArrays.Add(YValues);
```

Comment:
This is another copy that we don't need as well. We could either add an empty array to the end of the YValueArrays and populate that in the for loop, or we can do it as you're doing now and instead Move the array into the array of arrays.
```suggestion
		YValueArrays.Add(MoveTemp(YValues));
```

Reply:
Not sure I've explained that very well - let me know :)